Bow OpenAPI/Generator/Algebra/FileSystem.swift

```diff
@@ -37,4 +38,11 @@ extension FileSystem {
     func removeFiles(_ files: String...) -> IO<FileSystemError, ()> {
         files.traverse(remove(itemPath:)).void()^
     }
+    
+    func rename(_ name: String, itemAt: String) -> IO<FileSystemError, ()> {
+        let copyItem = copy(itemPath: itemAt, toPath: "\(itemAt.parentPath)/\(name).\(itemAt.extension)")
+        let removeCopiedItem = removeFiles(itemAt)
+        
+        return copyItem.followedBy(removeCopiedItem)^
+    }
```

Comment:
This implementation is being too clever about renaming and assuming you want to keep the same extension, which may not always be the case. Force the user to pass the name **with** the extension. It can even be defined in terms of a `move` function:

```swift
func move(from origin: String, to destination: String) -> IO<FileSystemError, Void> {
  copy(itemPath: origin, toPath: destination)
    .followedBy(removeFiles(origin))
}

func rename(_ newName: String, itemAt: String) -> IO<FileSystemError, Void> {
  move(from: itemAt, to: "\(itemAt.parentPath)/\(newName)")
}
```